### PR TITLE
fix flaky tests

### PR DESCRIPTION
**Pull request by matszczygiel**
This PR fixes a flakiness in some tests, plus some logic errors:
* Improve the reconnection logic, so that the trigger is not ignored while in the backoff loop
* There was a bug where the file was stopped for good on disconnection

I've also split the analytics tests from the regular ones, for easier inspection

### CODE REVIEW

File: drop-transfer/src/ws/client/mod.rs [hunk 1]
```diff
@@ -114,34 +115,14 @@ pub(crate) fn spawn(
...
-    ControlFlow::Continue(())
-}
-
 async fn connect_to_peer(
     state: &Arc<State>,
     xfer: &Arc<OutgoingTransfer>,
     logger: &Logger,
     alive: &AliveGuard,
 ) -> ControlFlow<()> {
+    debug!(logger, "Outcoming transfer job started for {}", xfer.id(),);
```
Comment (Lipt0nas) [inline drop-transfer/src/ws/client/mod.rs:124]:
```suggestion
    debug!(logger, "Outgoing transfer job started for {}", xfer.id(),);
```

File: test/scenarios.py [hunk 2]
```diff
@@ -10145,7 +10145,7 @@
         },
     ),
     Scenario(
-        "scenario-analytics-1",
+        "analytics-1",
```
Comment (Lipt0nas) [inline test/scenarios.py:10148]: I really like the idea of splitting analytics tests into a separate job

There is still a question for scenarios `26-1` and `26-2`, they are also testing analytics events, though they also have some other functional testing alongside it, i would lean towards moving them to the `analytics` space as well, what is your opinion?

File: drop-transfer/src/ws/server/mod.rs [hunk 3]
```diff
@@ -905,15 +905,7 @@ impl FileXferTask {
...
                         Err(crate::Error::Canceled) => {
-                            info!(logger, "File {} cancelled", self.file.id());
-
-                            if let Err(err) = state
-                                .transfer_manager
-                                .incoming_download_cancel(self.xfer.id(), self.file.id())
-                                .await
-                            {
-                                warn!(logger, "Failed to store download finish: {err}");
-                            }
```
Comment (Lipt0nas) [inline drop-transfer/src/ws/server/mod.rs:916]: Could you explain on how this was a bug?

**Review by Lipt0nas — COMMENTED** [review]
Really clean refactor of the wait loop :ok_hand:

File: test/scenarios.py (hunk 2 above)
Comment (matszczygiel) [reply test/scenarios.py:10149]: Yeah, sure. Good point :+1:

File: drop-transfer/src/ws/client/mod.rs (hunk 1 above)
Comment (matszczygiel) [reply drop-transfer/src/ws/client/mod.rs:124]: done

File: test/scenarios.py (hunk 2 above)
Comment (matszczygiel) [reply test/scenarios.py:10149]: Moved

File: drop-transfer/src/ws/server/mod.rs (hunk 3 above)
Comment (Lipt0nas) [reply drop-transfer/src/ws/server/mod.rs:916]: Ah i see, thank you for explaining

True about QA, perhaps they're still doing more surface tests

**Review by Lipt0nas — APPROVED** [review]
Damn is it nice to see dev pipelines being fully green (even if still slightly flaky)

**Pull request merged**